core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -137,6 +139,14 @@ public class GraphHopper {
     private String vehiclesString = "";
 
 
+    //for dynamic speeds
```

Comment:
Should we create a special `Graphhopper` instance like `GraphhopperCustomSpeeds` to add our custom logic? 
Basically for consistency with the logic I see Graphhopper team is following like in GraphhopperGtfs

On the other hand, probably, will make a bit cleaner the possible merge with Graphhopper 9? Also with a specific GraphhoperCustomSpeeds you can remove condition like `if (speedProvider != null)` as it will be mandatory for that GH instance

Reply:
> Should we create a special `Graphhopper` instance like `GraphhopperCustomSpeeds` to add our custom logic? Basically for consistency with the logic I see Graphhopper team is following like in GraphhopperGtfs
> 
> On the other hand, probably, will make a bit cleaner the possible merge with Graphhopper 9? Also with a specific GraphhoperCustomSpeeds you can remove condition like `if (speedProvider != null)` as it will be mandatory for that GH instance

Yes, you are right better to define a different instance